src/board.py

```diff
@@ -154,16 +154,21 @@ def capturable(source_piece: Piece, target: Square):
 
         return _moves
 
-    def move(self, source: Square | str, target: Square | str):
+    def move(self, source_piece: Piece | None, target: Square | str):
```

Comment:
This is what I was referring to, why call piece when it is `None`. Check in the calling code once instead of every function that takes as argument a piece.

Reply:
To make it clearer:
```python
add(x: int | None, y: int | None):
    if x is None or y is None
        return None
    return x + y
```
Would you do the above?

Reply:
This is a sound logic. To tell the truth, most None checks are there because Pylance is whining. I could add a type-ignore instead too, I suppose. I'll follow-up on your suggestion here and let you know ho many checks I trim.

Reply:
This remains to be implemented. As far as `Piece` code and type checking goes, the issue stems from the fact that we chose empty squares to be `None`. `Square` has a similar issue because I have the `Piece.square` attribute, which can be `None` to indicate a that a piece is “captured” or “orphan” (if created outside a board to begin with).